### PR TITLE
chore: add architecture overview for recent refactoring

### PR DESCRIPTION
**Pull request by Kavindu-Dodan**
## This PR

Adds a component diagram for flag configuration providers and their interactions with the runtime and other systems.

Related PR #291

### CODE REVIEW

File: docs/other_resources/high_level_architecture.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
 # High level architecture
 
+### Component overview
 <img src="../images/of-flagd-0.png" width="560">
+
+### Flag configuration provider
```
Comment (beeme1mr) [inline docs/other_resources/high_level_architecture.md:6]: Should these be H2s? Also, could you add a brief description under each section?

Comment (Kavindu-Dodan) [reply docs/other_resources/high_level_architecture.md:6]: Yeah they can be H2. I will update the sections with next commit :)

**Review by beeme1mr — APPROVED** [review]

**Comment by toddbaert** [comment]
I find the label "ISync Impl Listner" (sp) a bit misleading, probable because in code/comments, we refer to it something like "DataSync channel watcher"... That is what this represents, right? I think that might be a better name. Or just "DataSync Channel"? I think we should choose a consistent name for this abstraction.

**Comment by Kavindu-Dodan** [comment]
> I find the label "ISync Impl Listner" (sp) a bit misleading, probable because in code/comments, we refer to it something like "DataSync channel watcher"... That is what this represents, right? I think that might be a better name. Or just "DataSync Channel"? I think we should choose a consistent name for this abstraction.

I agree with this. I will rephrase this to "DataSync channel listener"

**Review by james-milligan — APPROVED** [review]

**Review by beeme1mr — APPROVED** [review]

**Review by toddbaert — APPROVED** [review]

**Comment by toddbaert** [comment]
@Kavindu-Dodan I meant to ask this before merge, but forgot... can you also commit the excalidraw source?
